airflow/models/connection.py

```diff
@@ -140,6 +140,7 @@ def __init__(
 
         if self.password:
             mask_secret(self.password)
+            mask_secret(quote(self.password))
```

Comment:
Why do we need to mask twice?

Reply:
We quote the password in the get uri method, [here](https://github.com/apache/airflow/blob/main/airflow/models/connection.py#L257) this will convert a password like "s3cr3t!" into "s3cr3t%21" converting the special chars into encoded chars. So when these parameters are accessed they don't get masked because they differ from the unquoted chars.